localization/ndt_scan_matcher/src/tree_structured_parzen_estimator.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright 2023 Autoware Foundation
```

Comment:
Could you add any reference to TPE so that we can review the internal algorithm as well? (Maybe something like https://papers.nips.cc/paper_files/paper/2011/file/86e8f7ab32cfd12577bc2619bc635690-Paper.pdf?)